core/modules/checkout/components/Shipping.ts

```diff
@@ -173,12 +174,20 @@ export const Shipping = {
       let currentShippingMethod = this.getCurrentShippingMethod()
       if (currentShippingMethod) {
         this.shipping = Object.assign(this.shipping, {shippingCarrier: currentShippingMethod.carrier_code})
-        this.$bus.$emit('checkout-after-shippingMethodChanged', {
-          country: this.shipping.country,
-          method_code: currentShippingMethod.method_code,
-          carrier_code: currentShippingMethod.carrier_code,
-          payment_method: this.paymentMethod[0].code
-        })
+        this.$store.dispatch('checkout/saveShippingDetails', this.shipping)
+          .then(() => {
+            const shippingMethodsData = createOrderData({
+              shippingDetails: this.$store.getters['checkout/getShippingDetails'],
+              shippingMethods: this.$store.getters['checkout/getShippingMethods'],
+              paymentMethods: this.$store.getters['checkout/getPaymentMethods'],
+              paymentDetails: this.$store.getters['checkout/getPaymentDetails']
+            })
+            Object.assign(shippingMethodsData, {
+              carrier_code: this.shipping.shippingCarrier,
+              method_code: this.shipping.shippingMethod
+            })
+            this.$bus.$emit('checkout-after-shippingMethodChanged', shippingMethodsData)
+          })
       }
```

Comment:
I think we need to update the useMyAddress method code as well
https://github.com/DivanteLtd/vue-storefront/blob/develop/core/modules/checkout/components/Shipping.ts#L122